rotkehlchen/rotkehlchen.py

```diff
@@ -903,3 +910,44 @@ def create_oracle_cache(
             return  # only for cryptocompare for now
 
         self.cryptocompare.create_cache(from_asset, to_asset, purge_old)
+
+    def query_missing_prices(
+        self,
+        filter_query: HistoryEventFilterQuery,
+    ) -> None:
+        """Queries missing prices for HistoryBaseEntry in database based
+        on the query filter used"""
+        # Use a deepcopy to avoid mutations in the filter query if it is used later
+        new_filter_query = copy.deepcopy(filter_query)
+        new_filter_query.filters.append(DBStringFilter(and_op=True, column='usd_value', value='0'))
+        assets_missing_assets = self.data.db.get_rows_missing_prices_in_base_entries(
+            filter_query=new_filter_query,
+        )
+        inquirer = PriceHistorian()
+        updates = []
+        for identifier, amount, asset_name, timestamp in assets_missing_assets:
+            try:
+                # Shouldn't raise DeserializationError since data comes from the database
+                amount = deserialize_fval(
+                    value=amount,
+                    name='historic base entry usd_value query',
+                    location='query_missing_prices',
+                )
+                price = inquirer.query_historical_price(
+                    from_asset=Asset(asset_name),
+                    to_asset=A_USD,
+                    timestamp=Timestamp(int(timestamp / KRAKEN_TS_MULTIPLIER)),
```

Comment:
Perhaps we can make sure this timestamp is returned divided and to the "our format" from the `get_rows_missing_prices_in_base_entries()`?

It's good to serialize/deserialize as early as possible to our format. Otherwise we forget where we have what.